### PR TITLE
LUCENE-9263: Fix wrong transformation of distance in meters to radians in Geo3DPoint

### PR DESCRIPTION
**Pull request by iverase**
This PR removes the calculation from the planet model and move it to Geo3DUtil. Planet model is generic and it does not need and spheroid representing the earth.

@nknize can you have a look? We should fix it before the 8.5 release?

### CODE REVIEW

**Comment by iverase** [comment]
@nknize, I had another iteration and decided to go back to one constructor that defines the semiMajorAxis and semiMinorAxis only (The fattening factor was not used and can be calculated anyway from those two values).

File: lucene/spatial3d/src/test/org/apache/lucene/spatial3d/TestGeo3DPoint.java [hunk 1]
```diff
@@ -1328,6 +1329,32 @@ public void testEncodeDecodeIsStable() throws Exception {
     }
   }
 
+  public void testDistanceQuery() throws Exception {
+    Directory dir = newDirectory();
+    RandomIndexWriter w = new RandomIndexWriter(random(), dir);
+
+    PlanetModel planetModel = PlanetModel.WGS84;
```
Comment (nknize) [inline lucene/spatial3d/src/test/org/apache/lucene/spatial3d/TestGeo3DPoint.java]: Can we test this with the PlanetModel randomly selected from all three (Clarke, Sphere, WGS84)?

Comment (iverase) [reply lucene/spatial3d/src/test/org/apache/lucene/spatial3d/TestGeo3DPoint.java]: done

**Comment by iverase** [comment]
One thing that worries me is that we built Planet models using minor and major axis but equality of a planet model is done through xyScaling and zScaling, is that an issue for you?

**Comment by nknize** [comment]
@iverase good point. With these changes the `equals` and `hashCode` methods should also be changed to use major and minor axis

**Comment by iverase** [comment]
updated

**Pull request merged**